scripts/check_compliance.py

```diff
@@ -588,6 +588,118 @@ def run(self):
             self.add_failure(msg.decode("utf-8"))
 
 
+class PyLint(ComplianceTest):
+    """
+    Runs pylint on all .py files, with a limited set of checks enabled
+    """
+    _name = "pylint"
+    _doc = "https://www.pylint.org/"
+
+    def run(self):
+        self.prepare(ZEPHYR_BASE)
+
+        # The commented-out checks below might be useful to enable, but
+        # currently trigger false positives
+        cmd = ("pylint --jobs=0 --score=n --persistent=n --disable=all "
+               "--enable "
+               "C0303,"  # Trailing whitspace
```

Comment:
Can you move this configuration to some `--rcfile=pylint-requirements`?

This may help get rid of the numerical IDs.

Reply:
Gonna make a commit later to move this repo back to `scripts/ci` in the main Zephyr repo.

Reply:
Ohh... you mean as an argument to `pylint`.

IMO, it's nicer to have directly in the code. Cuts down on the number of files and makes the list easy to discover.

One downside is that people might mess up the formatting, but maybe the simplicity outweighs that.

Reply:
Could maybe make it a triple-quoted string or the like to avoid that though, and `','.join(pylint_checks)` to automatically insert the `,`s. Maybe just a string is good enough though.

Reply:
It should be the same thing. The only thing that differs is in which file you edit the options.

There's a general trend of looking at anything that seems too simple and direct with suspicion in modern software dev. Stupidly simple is freaking awesome. :ok_hand:

Reply:
> The only thing that differs is in which file you edit the options.

If configuration and code are in the same file then you cannot simply "git checkout" only one of them and test. You cannot git log the config file to look at just the configuration history. You cannot run pylint without this script. You cannot easily compare your configuration to someone else or pass it / share it with some other tool. Etc., etc.

I really don't see what's complicated about separating configuration from the code, it's a standard (.checkpatch.conf, .gitlint, CODEOWNERS,...) and massively successful approach and I don't remember anyone complaining about it before.

Reply:
Could argue that \<anything> should be moved into a separate file that way though ("if X and Y are in the same file you cannot simply...").

Do you think it's common to need the whitelist from commit X but not the rest of the test (which is still simple to do)? Extra files by themselves are a liability, so nice to skip them otherwise.

> You cannot run pylint without this script

Why not run the test itself if you want to do what the test does? (Genuine question... might be something I've missed.)

> I really don't see what's complicated about separating configuration from the code, it's a standard (.checkpatch.conf, .gitlint, CODEOWNERS,...) and massively successful approach and I don't remember anyone complaining about it before.

Here you already know what the configuration is. It's just moving it into a separate file.

Whether it's a list of parameters to some external command or some other data associated with the test makes no difference to me.